### ISSUE TITLE
`TypeError: setting getter-only property "error"` on failed update

### ISSUE THREAD

**Issue by SillyFreak** [ujjwalguptaofficial/JsStore#137]
I get the error from the title when violating a unique constraint when updating one of my tables:

```
// schema of relevant table
const tblProjects = {
  name: 'Projects',
  columns: {
    id: {
      primaryKey: true,
      autoIncrement: true,
    },
    name: {
      dataType: 'string',
      notNull: true,
      unique: true,
    },
  },
};

// example code triggering the error
export async function mcve() {
  const conn = await connection;

  // clear table
  {
    const rows = await conn.remove({
      from: 'Projects',
    });
    console.log('deleted all', rows, 'rows');
  }

  // create projects
  const [foo, bar] = await conn.insert({
    into: 'Projects',
    values: [{ name: 'foo' }, { name: 'bar' }],
    return: true,
  });
  console.log('created projects', foo, bar);

  // rename bar project
  try {
    const { id, ...values } = bar;
    values.name = 'foo';
    console.log('update project', id, 'with', values);

    const rows = await conn.update({
      in: 'Projects',
      set: values,
      where: { id },
    });
    console.log('updated', rows, 'rows');
  } catch (ex) {
    console.log('update failed:', ex);
  }
}
```

This is the error output (Firefox):

```
deleted all 2 rows

created projects Object { name: "foo", id: 118 } Object { name: "bar", id: 119 }

update project 119 with Object { name: "foo" }

Object { message: "A mutation operation in the transaction failed because a constraint was not satisfied.", type: "ConstraintError" }

updated 1 rows

TypeError: setting getter-only property "error"
```

That last TypeError as reported in Chromium:

```
Uncaught TypeError: Cannot assign to read only property 'error' of object '#<IDBRequest>'
    at IDBRequest.Base.onErrorOccured 
```

(The location should be [src/worker/business/base.ts:42](https://github.com/ujjwalguptaofficial/JsStore/blob/master/src/worker/business/base.ts#L42))

As you can see, `conn.update()` doesn't fail and "updated 1 rows" is printed. The error is logged as always, but after `conn.update()` has already completed a TypeError is reported. (in Chromium, the TypeError is consistently reported before the "updated 1 rows".) I suspect the error on the worker was swallowed by the TypeError.

Expected behavior: the `update` should throw an exception.

**Comment by ujjwalguptaofficial** [comment]
@SillyFreak  thanks for great description. The issue has been fixed in new version - 3.4.3. Please test and confirm.

Thanks

**Comment by SillyFreak** [comment]
Yes, it works now as expected! Thanks for reacting so quickly!
